## Task 1

In [ ]:
import pandas as pd
print(pd.__version__)

## Task2

In [ ]:
df = pd.read_csv("data/customers.csv")
df

## Task3

In [ ]:
df.head(6)

In [ ]:
df.tail(4)

In [ ]:
df.sample(5)

## Task4

In [13]:
cars = pd.read_csv("data/cars.csv")
cars.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   car name      398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


- 398 Rows
- 9 Columns
- MPG, Cylinders, Displacement, Horsepower, Weight, Acceleration, Model Year, Origin, Car Name

## Task5
- Whole Numbers: Cylinders, Weight, Model Year, Origin
- Floats: MPG, Displacement, Acceleration
- Objects: Horsepower, Car Name

## Task6

In [14]:
titanic = pd.read_csv("data/titanic_survival.csv")
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


- 891 Rows, 15 columns
- Most missing values are in "deck"
- You know a value is missing by the "Non-Null Count" column, where the difference between the non-null count and the index range is the amount of missing values.

## Task7

In [20]:
cars.describe()
cars["horsepower"]

0      130
1      165
2      150
3      150
4      140
      ... 
393     86
394     52
395     84
396     79
397     82
Name: horsepower, Length: 398, dtype: object

- 23.51
- Data Type is an object for horsepower???
- 